File: Ambulance_Spring/src/main/java/com/example/ambulance_spring/entities/Mission.java
package com.example.ambulance_spring.entities;


import com.example.ambulance_spring.entities.enums.MissionStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "missions")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Mission {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;



    private double latitude;

    private double longitude;


    @ManyToOne
    @JoinColumn(name = "driver_id")
    private User driver;

    @ManyToOne
    @JoinColumn(name = "dispatcher_id")
    private User dispatcher;

    @ManyToOne
    @JoinColumn(name = "hospital_id")
    private Hospital hospital;


    @Enumerated(EnumType.STRING)
    private MissionStatus status;
}

File: dashboard/types/mission.ts
import { User } from "./user";
import { Hospital } from "./hospital";

export enum MissionStatus {
  PICKUP = "PICKUP",
  ONROUTETOHOSPITAL = "ONROUTETOHOSPITAL",
  DONE = "DONE",
}

export interface Mission {
  id: number;
  status: MissionStatus;
  latitude: number;
  longitude: number;
  driver: number;
  dispatcher: number;
  hospital: number;
}

